### PR TITLE
Render a collection list on the collection tab

### LINKED ISSUE #46: Render a collection list on the collection tab
**Is your feature request related to a problem? Please describe.**
When we switch over to the collection tab it should load a list of collection similar to the contacts

**Describe the solution you'd like**
Use the same search graphql api as we have used for contacts. The variables will be changed a bit

```
const variables = {
  filter: { searchGroup: true},
  messageOpts: {
    limit: 3,
    offset: 0,
  },
  contactOpts: {
    limit: 10,
    offset: 0,
  },
};
```

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ajaman190**
### Description:
This pull request addresses the issue #46 . The collection list will be displayed in a similar manner to the contacts list.

### Changes Made:
1. Modify the collection list component for rendering list with dynamic variables.
2. Added the necessary logic and API calls to fetch the collection data.
3. Integrated the collection list component into the collection tab view.

### Screenshots:

https://github.com/glific/mobile/assets/84832565/663e07a9-a280-4a02-b177-4e4a26bd1b58

### Closes Issue:
Closes #46 

Please review and merge this pull request. Thank you!

### CODE REVIEW

**Comment by mdshamoon** [comment]
@ajaman190 please make every new branch from main rather than your previous branch. It makes it difficult to review just the changes required here. Right now its showing 35 files which are from the previous issues.

I will look at the code later but based on the screenshot two things the names of the collection are present. You need to modify the search query as well. Soory missed mentioning in the issue. I thought it was already present

```
export const GET_COLLECTIONS = gql`
  query search($filter: SearchFilter!, $contactOpts: Opts!, $messageOpts: Opts!) {
    search(
      filter: $filter
      contactOpts: $contactOpts
      messageOpts: $messageOpts
    ) {
      group {
        name
      }
    }
  }
`;


```

These are all the fields that can be used but we will need them at a later stage

```
export const SEARCH_QUERY = gql`
  query search($filter: SearchFilter!, $contactOpts: Opts!, $messageOpts: Opts!) {
    search(filter: $filter, contactOpts: $contactOpts, messageOpts: $messageOpts) {
      contact {
        id
        name
        phone
        fields
        maskedPhone
        lastMessageAt
        status
        bspStatus
        isOrgRead
      }
      group {
        id
        label
      }
      messages {
        id
        body
        insertedAt
        messageNumber
        receiver {
          id
        }
        sender {
          id
        }
        location {
          latitude
          longitude
        }
        type
        media {
          url
          caption
        }
        errors
        contextMessage {
          body
          contextId
          messageNumber
          errors
          media {
            caption
            sourceUrl
            id
            url
          }
          type
          insertedAt
          location {
            id
            latitude
            longitude
          }
          receiver {
            id
          }
          sender {
            id
            name
          }
        }
        interactiveContent
        sendBy
        flowLabel
      }
    }
  }
`;

```

File: screens/Collections.tsx [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+
+type Props = NativeStackScreenProps<RootStackParamList, 'Contacts'>;
 
-const Collections = () => {
+const Collections = ({ navigation }: Props) => {
   return (
     <View style={styles.mainContainer}>
-      <SearchBar />
-      <Text>Collections</Text>
+      <ApolloProvider client={client}>
```
Comment (mdshamoon) [inline screens/Collections.tsx:21]: Lets move this ApolloProvider at the top level and not initialize in every component. Maybe App.tsx

File: components/ui/ContactList.tsx [hunk 2]
```diff
@@ -37,6 +41,7 @@ const ContactList: React.FC<any> = ({ navigation }: any) => {
   let contacts = [];
   if (data) {
     contacts = data.search.map((element: any, idx: number) => {
+      console.log(element)
```
Comment (mdshamoon) [inline components/ui/ContactList.tsx:44]: Remove console

File: screens/Collections.tsx (hunk 1 above)
Comment (ajaman190) [reply screens/Collections.tsx:21]: I was also thinking the same.

**Comment by ajaman190** [comment]
Screenshot_20230524-201212

I have also updated the ui little bit.

**Pull request merged**